coffee-chats/src/main/java/com/google/step/coffee/servlets/ChatRequestServlet.java

```diff
@@ -38,21 +38,31 @@ public Object post(JsonServletRequest request) throws IOException, HttpError {
 
   private ChatRequest buildChatRequestFromHttpRequest(HttpServletRequest request) throws HttpError {
     List<String> tags = getParameterValues("tags", request);
-    List<String> dateStrings = getParameterValues("dates", request);
+    List<String> startDateStrings = getParameterValues("startDates", request);
+    List<String> endDateStrings = getParameterValues("endDates", request);
     int minPeople = parseIntParam("minPeople", request);
     int maxPeople = parseIntParam("maxPeople", request);
     int durationMins = parseIntParam("durationMins", request);
     boolean matchRandom = parseBooleanParam("matchRandom", request, "false");
     boolean matchRecents = parseBooleanParam("matchRecents", request, "true");
 
-    List<Date> dates = dateStrings.stream()
+    List<Date> startDates = startDateStrings.stream()
         .map(Long::parseLong)
         .map(Date::new)
         .collect(Collectors.toList());
 
+    List<Date> endDates = endDateStrings.stream()
+        .map(Long::parseLong)
```

Comment:
Ideally you'd want to catch `NumberFormatException` here

Reply:
Good point, I have added error handling for this in the servlet.